pkg/trace/api/telemetry.go

```diff
@@ -0,0 +1,153 @@
+
+	m.Director(req)
+
+	resp, err := m.Transport.RoundTrip(req)
+	if err != nil {
+		metrics.Count("datadog.trace_agent.telemetry_proxy.roundtrip.errors", 1, tags, 1)
+	}
+	return resp, err
+}
+
+// NewReverseProxy creates an http.ReverseProxy which will forward requests via transport
+func NewReverseProxy(transport http.RoundTripper, logger *stdlog.Logger) http.Handler {
+	director := func(req *http.Request) {
+		req.Header.Set("Via", fmt.Sprintf("trace-agent %s", info.Version))
+		if _, ok := req.Header["User-Agent"]; !ok {
+			// explicitly disable User-Agent so it's not set to the default value
+			// that net/http gives it: Go-http-client/1.1
+			// See https://codereview.appspot.com/7532043
+			req.Header.Set("User-Agent", "")
+		}
+	}
+	return &httputil.ReverseProxy{
+		Director:  director,
+		ErrorLog:  logger,
+		Transport: transport,
+	}
+}
+
+// RoundTrip sends request first to MainTarget, then sends a copy of main request to every configurged
+// AdditionalTargets receiptient.
+//
+// All requests will be sent irregardless of any errors
+// If any request fails, the error will be logged. Only main target's
+// error will be propagated via return value
+func (m *telemetryMultiTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	if len(m.Endpoints) == 1 {
+		return m.roundTrip(req, m.Endpoints[0])
+	}
+	slurp, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		return nil, err
+	}
+	newreq := req.Clone(req.Context())
+	newreq.Body = ioutil.NopCloser(bytes.NewReader(slurp))
+	rresp, rerr := m.roundTrip(newreq, m.Endpoints[0])
+	for _, endpoint := range m.Endpoints[1:] {
+		newreq := req.Clone(req.Context())
+		newreq.Body = ioutil.NopCloser(bytes.NewReader(slurp))
+		if resp, err := m.roundTrip(newreq, endpoint); err == nil {
+			// we discard responses for all subsequent requests
+			io.Copy(ioutil.Discard, resp.Body)
+			resp.Body.Close()
+		} else {
+			log.Error(err)
+		}
+	}
+	return rresp, rerr
+}
+
+// telemetryProxyHandler parses returns a new HTTP handler which will proxy requests to the configured intakes.
+// If the main intake URL can not be computed because of config, the returned handler will always
+// return http.StatusInternalServerError along with a clarification.
+func (r *HTTPReceiver) telemetryProxyHandler() http.Handler {
+	if !r.conf.TelemetryConfig.Enabled {
+		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			msg := fmt.Sprintf("Telemetry proxy forwarder is Disabled")
+			http.Error(w, msg, http.StatusMethodNotAllowed)
+		})
+	}
+
+	// extract and validate Hostnames from configured endpoints
+	var endpoints []*config.Endpoint
+	for _, endpoint := range r.conf.TelemetryConfig.Endpoints {
+		u, err := url.Parse(endpoint.Host)
+		if err != nil {
+			log.Errorf("Error parsing apm_config.telemetry endpoint: %s", endpoint.Host)
+			continue
+		}
+		if u.Host != "" {
+			endpoint.Host = u.Host
+		}
+
+		endpoints = append(endpoints, endpoint)
+	}
+	if len(endpoints) == 0 {
```

Comment:
Since I removed previous handler - now that IsEnabled is used, I think a special handler here would be useful.
I can make the message more generic though.